src/AWS.Deploy.Orchestration/Utilities/TemplateMetadataReader.cs

```diff
@@ -56,67 +56,29 @@ private static CloudApplicationMetadata ReadSettings(string templateBody)
         {
             try
             {
-                var metadataSection = ExtractMetadataSection(templateBody);
-
-                var yamlMetadata = new YamlStream();
-                using var reader = new StringReader(metadataSection);
-                yamlMetadata.Load(reader);
-                var root = (YamlMappingNode)yamlMetadata.Documents[0].RootNode;
-                var metadataNode = (YamlMappingNode)root.Children[new YamlScalarNode("Metadata")];
+                var cfTemplate = JsonConvert.DeserializeObject<CFTemplate>(templateBody);
```

Comment:
You need to detect if the template is either JSON or YAML and then use the appropriate parser.

Reply:
Done